components/Hamburger/index.js

```diff
@@ -5,15 +5,19 @@ import React, {Component, PropTypes} from 'react';
 class Hamburger extends Component {
   render(): ?ReactElement {
     var className = 'Hamburger';
-    var {target, x, onClick} = this.props;
+    var {target, onClick} = this.props;
+    var {topOfViewport, navVisible} = this.context;
```

Comment:
first time using the undocumented feature of react: context. I gotta admit that was a magical experience.